netmiko/supermicro/smci_base_connection.py

```diff
@@ -0,0 +1,187 @@
+"""SmciBaseConnection is netmiko SSH class for Supermicro blade/ToR switch platforms."""
+from netmiko.base_connection import BaseConnection
+from netmiko.ssh_exception import NetmikoAuthenticationException
+import re
+import time
+
+
+class SmciBaseConnection(BaseConnection):
```

Comment:
Why not inherit from CiscoBaseConnection? Can we get rid of or greatly simplify the serial_login and telnet_login code if we do this?

Reply:
The reason we didn’t inherit CiscoBaseConnection class is because there are some differences between cisco and our cli. After internal discussion, we’d like to take your suggestions to inherit CiscoBaseConnection directly and override some methods for dealing with those differences. We are working on it now.